### ISSUE TITLE
[jss-nested] search '&' in all the string

### ISSUE THREAD

**Issue by b2whats** [cssinjs/jss#297]
Example http://codepen.io/b2whats/pen/RRzwJR

We have two elements - 
1) input - functional component
2) underline - design component

Design component is optional, it is standalone, its styles should not be in the input file because it violates the principles of dry

``` js
// input.js
<div>
  <input class='input'>
  <Underline>
</div>

const styles = {
  input: {
    input styles,
    '&:focus + ????': {
      active underline style 
    }
  }
}

// underline.js

<hr class='underline'>

const styles = {
  underline: {
    hr styles,
    'input:focus + &': {
      active underline style 
    }
  }
}

```

change this[line](https://github.com/cssinjs/jss-nested/blob/master/src/index.js#L36)

``` js
'string with &'.indexOf('&')
```

**Comment by kof** [comment]
1. Matching & everywhere in the string will slow down a bit performance, as it needs to be done on every property. (not sure how much overhead it is, needs benchmarks)
2. It allows you to create a "non-scoped" selector like "input hr" which is what you normally do with "named: false" and considered as legacy/bad style.


I have 3 suggestions  how to solve this:

1. Create an InputWithUnderline component which uses "input" and "Underline" components. It can put the class name on input and generate the sibling selector as you wanted.

```js
const styles = {
  underline:  {
    ...underline styles
  },
  input: {
    '&:focus + $underline': {
      active underline style 
    }
  }
}
```

2. Use javascript events and set proper classes.

**Comment by kof** [comment]
Released with jss-nested@2.5.0
